### PR TITLE
misc(build): inline report generator resources for devtools

### PR DESCRIPTION
**Pull request by connorjclark**
- Remove `REPORT_CSS` from `report-assets.js`, as nothing uses it directly, except for the `flow-report-assets.js`, but that now reads the file itself. This was done so that...
- The `report-generator.js` build script no longer needs a separate module for the `report-assets.js`: instead we can just shim the `flow-report-assets.js` module to drop them from the bundle, and rely on our inline plugin to do the job that this build script + legacy CDT modules was doing
- Use rollup instead of browserify to create the `report-generator.js` bundle

I also updated the documentation for how the roll into DevTools works, as quite a few things have changed since it was written. Please give that a close look and suggest anything to improve clarity or correctness.

Resolves the need to workaround legacy CDT module usages (see https://chromium-review.googlesource.com/c/devtools/devtools-frontend/+/3289316 )

### CODE REVIEW

File: report/generator/flow-report-assets.js [hunk 1]
```diff
@@ -9,12 +9,13 @@ const fs = require('fs');
...
 const FLOW_REPORT_TEMPLATE = fs.readFileSync(`${__dirname}/../../flow-report/assets/standalone-flow-template.html`, 'utf8');
+const REGULAR_REPORT_CSS = fs.readFileSync(__dirname + '/../assets/styles.css', 'utf8');
 const FLOW_REPORT_CSS = fs.readFileSync(`${__dirname}/../../flow-report/assets/styles.css`, 'utf8');
 const FLOW_REPORT_JAVASCRIPT = fs.readFileSync(`${__dirname}/../../dist/report/flow.js`, 'utf8');
 /* eslint-enable max-len */
 
 module.exports = {
   FLOW_REPORT_TEMPLATE,
-  FLOW_REPORT_CSS,
+  FLOW_REPORT_CSS: [REGULAR_REPORT_CSS, FLOW_REPORT_CSS].join('\n'),
```
Comment (adamraine) [inline report/generator/flow-report-assets.js:19]: I'm fine with this approach for now, but it's doing the same thing as https://github.com/GoogleChrome/lighthouse/pull/13260 which adds regular report styles via the inlined JS component.

File: build/readme.md [hunk 2]
```diff
@@ -18,37 +18,36 @@ To build the devtools files and roll them into a local checkout of Chromium:
...
+└── report
+    └── bundle.esm.js
 ```
 
-1. the big `lighthouse-dt-bundle.js` bundle
-1. the much smaller `report-generator.js` bundle (just two modules). This is exported as ReportGenerator
-1. copies all the `report.{js,css}` / `template(s).html` files (these are not transformed in any way). We call these the report assets.
+1. the biggest file is `lighthouse-dt-bundle.js`. This is a bundle of `lighthouse-core`, and is run inside a worker in CDT.
+1. the much smaller `report-generator.js` bundle. This is assigned to the global object as `Lighthouse.ReportGenerator`
+  - This bundle has inlined the `dist/report/standalone.js` and `standalone-template.html` files (these are not transformed in any way). We call these the report generator assets.
```
Comment (adamraine) [inline build/readme.md:35]: I'm wasn't seeing the indent

```suggestion
    - This bundle has inlined the `dist/report/standalone.js` and `standalone-template.html` files (these are not transformed in any way). We call these the report generator assets.
```

File: build/build-dt-report-resources.js [hunk 3]
```diff
@@ -29,16 +27,31 @@ function writeFile(name, content) {
 fs.rmSync(distDir, {recursive: true, force: true});
 fs.mkdirSync(distDir, {recursive: true});
 
-writeFile('report.js', htmlReportAssets.REPORT_JAVASCRIPT);
-writeFile('standalone-template.html', htmlReportAssets.REPORT_TEMPLATE);
+writeFile('report.js', '// This can be removed after the next CDT roll deletes this file');
+writeFile('standalone-template.html',
+  '<!-- This can be removed after the next CDT roll deletes this file -->');
 writeFile('report.d.ts', 'export {}');
 writeFile('report-generator.d.ts', 'export {}');
```
Comment (adamraine) [inline build/build-dt-report-resources.js:34]: Could you add a note about these in the readme?

**Review by adamraine — APPROVED** [review]

**Comment by TimvdLippe** [comment]
This is great, thanks for doing that! I would love to remove the final two `module.json` files from DevTools next week. I am not sure if you were already planning on doing so, but do you mind syncing LightHouse 9.1.0 to DevTools next week to unblock the removal? Thanks in advance!

**Pull request merged**